#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From c2b62b18597967ebd5d7b62022f811ab59ca538f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ingo=20Kl=C3=B6cker?= <dev@ingo-kloecker.de>
Date: Tue, 18 Feb 2025 14:11:35 +0100
Subject: [PATCH 2/2] Allow to use different icons for a customized build

By defining KLEOPATRA_ICON_DIR one can replace the default app icon
with a different app icon for Kleopatra.

GnuPG-bug-id: 7509
---
 src/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4afab7b59..fe6310bb5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -582,7 +582,11 @@ kconfig_add_kcfg_files(_kleopatra_SRCS
   kcfg/tooltippreferences.kcfgc
 )

-file(GLOB ICONS_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/icons/*-apps-kleopatra.*")
+# setting a different KLEOPATRA_ICON_DIR allows to use different icons for a customized build
+if (NOT KLEOPATRA_ICON_DIR)
+    set(KLEOPATRA_ICON_DIR "${CMAKE_CURRENT_SOURCE_DIR}/icons")
+endif()
+file(GLOB ICONS_SRCS "${KLEOPATRA_ICON_DIR}/*.png" "${KLEOPATRA_ICON_DIR}/*.svg*")
 ecm_add_app_icon(_kleopatra_SRCS ICONS ${ICONS_SRCS})
 ecm_install_icons(ICONS ${ICONS_SRCS} DESTINATION ${KDE_INSTALL_ICONDIR})

--
2.48.1
